### PR TITLE
Fixes #877 Add git http access flag, add cache repo secure flags

### LINKED ISSUE #877: Cache warmer does not support insecure registries
**Actual behavior**
Attempting to warm cache with an image from a private http registry fails with the following error: 
```
Failed warming cache: Failed to retrieve image: <registry_url>/<repo>:<tag>: Get https://<registry_url>/v2/: http: server gave HTTP response to HTTPS client
```

Attempting to use the `--insecure` or `--insecure-pull` options like the executor offers results in:
```
Error: unknown flag: --insecure
```
```
Error: unknown flag: --insecure-pull
```

**Expected behavior**
Cache warmer to have the same insecure registry support as executor.

**To Reproduce**
1. Run cache warmer to pull image from insecure registry
2. Run cache warmer again, attempting to use same insecure flags available in executor

**Additional Information**
- v0.14.0
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [x] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

### PR DESCRIPTION
**Pull request by nonetheless**
Fixes `#877`. _in case of a bug fix, this should point to a bug and any other related issue(s)_

**Description**
Fixes #877
Add git-insecure flag for local git server 
Move secure config to a struct, and add the flags to warmer cmd:
--insecure,  --insecure-pull, --insecure-registry,  --skip-tls-verify, --skip-tls-verify-pull, --skip-tls-verify-registry,

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [x] The code flow looks good. 
- [x] Unit tests and or integration tests added.

**Release Notes**

Describe any changes here so maintainer can include it in the release notes, or delete this block.

```
Examples of user facing changes:
- Skaffold config changes like
  e.g. "Add buildArgs to `Kustomize` deployer skaffold config."
- Bug fixes
  e.g. "Improve skaffold init behaviour when tags are used in manifests"
- Any changes in skaffold behavior
  e.g. "Artiface cachine is turned on by default."

```

### CODE REVIEW

**Comment by nonetheless** [comment]
@googlebot I signed it!

File: pkg/buildcontext/git_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/config"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+)
+
+func TestGitClone(t *testing.T) {
```
Comment (cvgw) [inline pkg/buildcontext/git_test.go]: Please move this test to the integration test suite `pkg/integration` as it requires specific configuration on the host machine.

File: pkg/warmer/warm_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/config"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+)
+
+func TestInsecureRegistry(t *testing.T) {
+	gopath := os.Getenv("GOPATH")
```
Comment (cvgw) [inline pkg/warmer/warm_test.go]: Please modify this test to utilize a temp dir

File: pkg/warmer/warm_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/config"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+)
+
+func TestInsecureRegistry(t *testing.T) {
```
Comment (cvgw) [inline pkg/warmer/warm_test.go]: Please move this test to the integration test suite

File: pkg/buildcontext/git_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	testGit := Git{
+		context: "github.com/GoogleContainerTools/kaniko.git#refs/heads/master",
+		options: &config.ContextOptions{
+			InsecureGit: false,
+		},
+	}
+	gopath := os.Getenv("GOPATH")
```
Comment (cvgw) [inline pkg/buildcontext/git_test.go]: Please update this to use a temp dir

File: pkg/executor/build.go [hunk 5]
```diff
@@ -627,7 +627,7 @@ func fetchExtraStages(stages []config.KanikoStage, opts *config.KanikoOptions) e
 			}
 			// This must be an image name, fetch it.
 			logrus.Debugf("Found extra base image stage %s", c.From)
-			sourceImage, err := util.RetrieveRemoteImage(c.From, opts)
+			sourceImage, err := util.RetrieveRemoteImage(c.From, &opts.SecureOptions)
```
Comment (cvgw) [inline pkg/executor/build.go]: `opts.SecureOptions` is being passed as a pointer in many places where I don't see any modifications happening to the object. Is there an advantage to passing it as a pointer when it is such a small struct?

File: pkg/buildcontext/git_test.go (hunk 1 above)
Comment (nonetheless) [reply pkg/buildcontext/git_test.go]: Thanks for review, I will change it

File: pkg/executor/build.go (hunk 5 above)
Comment (nonetheless) [reply pkg/executor/build.go]: emm，It seem be unnessasery， I will change it

Comment (nonetheless) [reply pkg/executor/build.go]: opts.SecureOptions is being passed as struct now

File: pkg/buildcontext/git_test.go (hunk 4 above)
Comment (nonetheless) [reply pkg/buildcontext/git_test.go]: I don't have a gcloud config, so I don't add it to integration

**Comment by cvgw** [comment]
@nonetheless let me know when you're ready for another review 👍

**Comment by jlemm45** [comment]
@nonetheless Any reason this wasn't merged? I'm looking to pull from an insecure git repo.

**Pull request closed without merging**